editor.planx.uk/src/@planx/components/shared/Schema/model.ts

```diff
@@ -118,10 +118,21 @@ export interface Schema {
   max?: number;
 }
 
+/**
+ * Value returned per field, based on field type
+ */
+export type ResponseValue<T extends Field> = T extends MapField
+  ? Feature[]
+  : T extends ChecklistField
+  ? string[]
+  : T extends NumberField
+  ? number
+  : string;
```

Comment:
Unfortunately the only way to make these conditional types are long ternaries like this.

There is an alternative of using [a mapped type](https://www.typescriptlang.org/docs/handbook/2/mapped-types.html), but I tried a few variations on this and to me it read as more complex than this.